src/lib/replication/service/generic_repl_svc.cpp

```diff
@@ -68,17 +69,22 @@ AsyncReplResult< shared< ReplDev > > GenericReplService::create_repl_dev(uuid_t
         }
     }
 
-    // Create whatever underlying implementation of repl_dev needs to be for fresh creation of repl_dev
-    auto const result = create_replica_set(group_id, std::move(members)).get();
-    if (!bool(result)) { return make_async_error< shared< ReplDev > >(result.error()); }
-
     // Now we need to create local repl dev instance which is actually creates entry, state machine etc
     // according the the underlying implementation
     superblk< repl_dev_superblk > rd_sb{get_meta_blk_name()};
     rd_sb.create(rd_super_blk_size());
     rd_sb->group_id = group_id;
 
     auto repl_dev = create_local_repl_dev_instance(rd_sb, false /* load_existing */);
+
```

Comment:
I think you are worried about rd_sb being copied inside the `create_local_repl_dev_instance` and this and other method are working on 2 different in-memory versions, right? If so that is an excellent catch. 

I will disable copy constructor and `=` operator on superblk_handler as it can happen even otherwise. So will do a move only and we shall move the superblk and access them from repl_dev->superblk() as reference. In that case, cp_flush() will not do rd_sb write() until this method finishes rd_sb.write() as repl_dev is not added into the map yet.

Reply:
yes, it will have 2 different in-memory versions, but that is not the point I was worried.  what I worries is two different version super_block will allocate 2 metablks on HS meta service when they are called with write() and we need recovery twice during restart.  that is because member m_meta_mgr_cookie in the both super_block is still nullptr and sb.write() function will allocate metablk instead of using the exist one.